### PR TITLE
[Reco] Fixes needed to avoid exposing private headers via interface dir

### PR DESCRIPTION
**Pull request by smuzaffar**
These fixes are needed to avoid exposing private headers files (files in non-interface directory of package) via public interface files
- `RecoLocalMuon/RPCRecHit`: It generates a plugin so all files from `interface/src` are moved to plugins directory
- `RecoLocalCalo/HcalRecAlgos`:
  - `src/HybridMinimizer.h` is needed by `interface/PulseShapeFitOOTPileupCorrection.h`
- `RecoPixelVertexing/PixelTriplets`:
  - `src/CACut.h`  and `src/CAGraph.h` are used by headers in `interface` direcotry. So move these to `interface` too

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

File: RecoPixelVertexing/PixelTriplets/interface/CACut.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #define RecoPixelVertexing_PixelTriplets_src_CACut_h
 
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "CAGraph.h"
```
Comment (slava77) [inline RecoPixelVertexing/PixelTriplets/interface/CACut.h:15]:
```suggestion
#include " RecoPixelVertexing/PixelTriplets/interface/CAGraph.h"
```

File: RecoPixelVertexing/PixelTriplets/interface/CACut.h [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #define RecoPixelVertexing_PixelTriplets_src_CACut_h
```
Comment (slava77) [inline RecoPixelVertexing/PixelTriplets/interface/CACut.h:11]:
```suggestion
#define RecoPixelVertexing_PixelTriplets_CACut_h
```
perhaps the same update can be done in RecoPixelVertexing/PixelTriplets/interface/CAGraph.h as well

File: RecoPixelVertexing/PixelTriplets/interface/CACut.h (hunk 1 above)
Comment (smuzaffar) [reply RecoPixelVertexing/PixelTriplets/interface/CACut.h]: done

File: RecoPixelVertexing/PixelTriplets/interface/CACut.h (hunk 2 above)
Comment (smuzaffar) [reply RecoPixelVertexing/PixelTriplets/interface/CACut.h]: done

**Comment by smuzaffar** [comment]
please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34919	5e57437
- code changes are technical, in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no relevant differences

**Comment by perrotta** [comment]
+1
- Technical

**Pull request merged**